library/src/main/java/com/datatorrent/lib/io/jms/JMSBase.java

```diff
@@ -76,18 +76,52 @@
   private transient Session session;
   private transient Destination destination;
 
-  private String connectionFactoryClass;
-  private Map<String, String> connectionFactoryProperties = Maps.newHashMap();
+  private ConnectionFactoryBuilder connectionFactoryBuilder = new DefaultConnectionFactoryBuilder();
   private String ackMode = "CLIENT_ACKNOWLEDGE";
-  private String clientId = "TestClient";
-  private String subject = "TEST.FOO";
+  private String clientId;
+  private String subject;
   private int batch = 10;
   private int messageSize = 255;
   private boolean durable = false;
   private boolean topic = false;
   private boolean verbose = false;
   protected boolean transacted = true;
 
+  public abstract static class ConnectionFactoryBuilder
+  {
+    protected Map<String, String> connectionFactoryProperties = Maps.newHashMap();
+
+    public Map<String, String> getConnectionFactoryProperties()
+    {
+      return connectionFactoryProperties;
+    }
+
+    public void setConnectionFactoryProperties(Map<String, String> connectionFactoryProperties)
+    {
+      this.connectionFactoryProperties = connectionFactoryProperties;
+    }
+
+    public abstract ConnectionFactory buildConnectionFactory();
+  }
+
+  public static class DefaultConnectionFactoryBuilder extends ConnectionFactoryBuilder
+  {
+
+    @Override
+    public ConnectionFactory buildConnectionFactory()
+    {
+      ConnectionFactory cf;
+      try {
```

Comment:
With the new scheme of using a ConnectionFactoryBuilder we don't need it (as we discussed it last). So the caller can just provide a custom builder with his own connectionFactoryClass in that builder.